### ISSUE TITLE
Is asyncify reentrancy always unsafe?

### ISSUE THREAD

**Issue by nhooyr** [emscripten-core/emscripten#10384]
https://emscripten.org/docs/porting/asyncify.html#reentrancy

> It is not safe to start an async operation while another is already running. The first must complete before the second begins.

Why is it unsafe to start an async operation while another one is running. Couldn't the stack of the second be saved as well and then resumed appropriately as needed?

I understand if your application cannot handle it but if your code does not rely on globals, it should work right?

cc @kripken

**Comment by kripken** [comment]
It could be supported in principle, yes. The current implementation doesn't allow it yet, for simplicity we started with the simple case. As I wrote in #10375 I agree it would be interesting to investigate that.

Closing this as a duplicate of that one, to keep the discussion in one place (feel free to respond there!).

**Comment by nhooyr** [comment]
Awesome, thank you <3
